qiskit/aqua/operators/state_fns/circuit_state_fn.py

```diff
@@ -277,6 +277,11 @@ def assign_parameters(self, param_dict: dict) -> OperatorBase:
     def eval(self,
              front: Union[str, dict, np.ndarray,
```

Comment:
I am not sure why, unlike OperatorBase, this has never been typehinted as Optional. Clearly it defaulted to None so should have been. Can we change this in all these state fn signatures especially now since it seem it will work in all cases whereas before it looks like things may well have failed in some of these should None have actually been passed.

Also I note the signature in OperatorBase for eval is a bit more specific around the Dict type on the eval signature than was done in the typehints in these state fns.

 Now circuit_state_fn, if I am reading the code correctly would already drop through and do what is coded here anyway as the default last line. Maybe you want this explicit code here so its consistent with the other state fns.